installinator/src/dispatch.rs

```diff
@@ -336,10 +336,15 @@ impl InstallOpts {
             )
             .register();
 
-        engine.execute().await.context("failed to execute installinator")?;
+        // Wait for both the engine to complete and all progress reports to be
+        // sent, then possibly return an error to our caller if either failed
+        // (but we want both to run to completion first, e.g. so that an engine
+        // failure is still reported to wicketd).
```

Comment:
Could you add a notation here about why we don't use `try_join!`? (It's due to the fact that one of the futures will be cancelled if the other exits).

Reply:
Definitely - reworded in [346654b](https://github.com/oxidecomputer/omicron/pull/3429/commits/346654b6d9e74a01578cc5ebe6d6a36218bb9785)